valhalla/thor/dijkstras.h

```diff
@@ -194,29 +234,186 @@ class Dijkstras {
+      }
+
+      // Check if the edge is allowed or if a restriction occurs
+      EdgeStatus* todo = nullptr;
+      uint8_t restriction_idx = -1;
+      if (offset_time.valid) {
+        // With date time we check time dependent restrictions and access
+        if (expansion_direction == ExpansionType::forward) {
+          if (!costing_->Allowed(directededge, pred, tile, edgeid, offset_time.local_time,
+                                 nodeinfo->timezone(), restriction_idx) ||
+              costing_->Restricted(directededge, pred, bdedgelabels_, tile, edgeid, true, todo,
+                                   offset_time.local_time, nodeinfo->timezone())) {
+            continue;
+          }
+        } else if (expansion_direction == ExpansionType::reverse) {
+          if (!costing_->AllowedReverse(directededge, pred, opp_edge, t2, opp_edge_id,
+                                        offset_time.local_time, nodeinfo->timezone(),
+                                        restriction_idx) ||
+              costing_->Restricted(directededge, pred, bdedgelabels_, tile, edgeid, false, todo,
+                                   offset_time.local_time, nodeinfo->timezone())) {
+            continue;
+          }
+        }
+      } else {
+        if (expansion_direction == ExpansionType::forward) {
+          if (!costing_->Allowed(directededge, pred, tile, edgeid, 0, 0, restriction_idx) ||
+              costing_->Restricted(directededge, pred, bdedgelabels_, tile, edgeid, true)) {
+            continue;
+          }
+        } else if (expansion_direction == ExpansionType::reverse) {
+          if (!costing_->AllowedReverse(directededge, pred, opp_edge, t2, opp_edge_id, 0, 0,
+                                        restriction_idx) ||
+              costing_->Restricted(directededge, pred, bdedgelabels_, tile, edgeid, false)) {
+            continue;
+          }
+        }
+      }
+
+      // Compute the cost and path distance to the end of this edge
+      sif::Cost transition_cost, newcost;
+      uint8_t flow_sources;
+
+      if (expansion_direction == ExpansionType::forward) {
+        transition_cost = costing_->TransitionCost(directededge, nodeinfo, pred);
+        newcost = pred.cost() +
+                  costing_->EdgeCost(directededge, tile, offset_time.second_of_week, flow_sources) +
+                  transition_cost;
+      } else if (expansion_direction == ExpansionType::reverse) {
+        transition_cost =
+            costing_->TransitionCostReverse(directededge->localedgeidx(), nodeinfo, opp_edge,
+                                            opp_pred_edge, pred.has_measured_speed(),
+                                            pred.internal_turn());
+        newcost =
+            pred.cost() + costing_->EdgeCost(opp_edge, t2, offset_time.second_of_week, flow_sources);
+        // TODO(danpat): why is this done differently in the reverse direction?
+        newcost.cost += transition_cost.cost;
```

Comment:
Does anyone know why the reverse implementation does this separate addition of the `.cost` field, rather than doing the same as the forward search does?

Reply:
@kevinkreiser Do you have any clue on this one?  Do these additions need to be done differently ?

Reply:
yeah so i see that in the reverse direction we only add the cost of the transition to the costing object but not the elapsed time of the transition... off the top of my head i dont know why that is... i'll have to look around at the code a bit more tonight to refresh m y memory unless @dnesbitt61 remembers?

Reply:
more succinctly, why in the forward direction to we accumulate both cost and secs for transitions: 
https://github.com/valhalla/valhalla/blob/master/src/thor/dijkstras.cc#L182-L184

but in the reverse we just take the cost:
https://github.com/valhalla/valhalla/blob/master/src/thor/dijkstras.cc#L370-L372

Reply:
Not sure why. Historically, the reverse direction required separating transition cost so that proper costing at the connection could be computed in FormPath. So it could have been omitted inadvertently from the reverse path when that was first changed.

Reply:
Ok - I tried consolidating the logic here and just doing `+ transition_cost` instead of _just_ the `.cost` field.

I ran:

```
./run_with_server.py --test-file ../test_requests/city_to_city/Australia_routes.txt  --concurrency 1
```

and observed no different in the routes output, indicating no impact on reachability calculations.  I'm going to repeat this for reverse isochrones, and if it has no impact, I'm inclined to remove this difference and consolidate the forward/reverse behaviour here.

Reply:
Turns out we don't have reverse isochrones anyway :-)

My best guess based on testing and these comments is that the existence of this difference is an overlooked bug from a past age in the code, rather than a difference in forward/reverse traversal.